Tests/FileIO/TestTetGenInterface.cpp

```diff
@@ -0,0 +1,110 @@
+/**
+ * \file   TestTetGenInterface.cpp
+ * \author Karsten Rink
+ * \date   2016-02-19
+ *
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ *
+ */
+
+#include <limits>
+#include <vector>
+#include <memory>
+
+#include "gtest/gtest.h"
+
+#include "BaseLib/BuildInfo.h"
+#include "FileIO/readMeshFromFile.h"
+#include "FileIO/TetGenInterface.h"
+#include "GeoLib/GEOObjects.h"
+#include "MeshLib/Mesh.h"
+#include "MeshLib/Node.h"
+#include "MeshLib/MeshInformation.h"
+
+// read TetGen geometry
+TEST(FileIO, TetGenSmeshReader)
+{
+	std::string const file_name (BaseLib::BuildInfo::data_path + "/FileIO/twolayermdl.smesh");
+	GeoLib::GEOObjects geo_objects;
+	FileIO::TetGenInterface tgi;
+	bool const result (tgi.readTetGenGeometry(file_name, geo_objects));
+	ASSERT_TRUE(result);
+
+	std::vector<GeoLib::Point*> const& pnts (*geo_objects.getPointVec("twolayermdl"));
+	ASSERT_EQ(744, pnts.size());
+	std::vector<GeoLib::Surface*> const& sfcs (*geo_objects.getSurfaceVec("twolayermdl"));
+	ASSERT_EQ(5, sfcs.size());
+	ASSERT_EQ(468, sfcs[2]->getNTriangles());
+	ASSERT_EQ(191, sfcs[3]->getNTriangles());
+}
+
+// existing mesh to TetGen geometry
+TEST(FileIO, TetGenSmeshInterface)
+{
+	std::string const file_name (BaseLib::BuildInfo::data_path + "/FileIO/AmmerSubsurfaceCoarse.vtu");
+	std::unique_ptr<MeshLib::Mesh const> const mesh (FileIO::readMeshFromFile(file_name));
+
+	std::string const tg_new_name ("TestSmeshWriter");
+	std::string const output_name(BaseLib::BuildInfo::tests_tmp_path + tg_new_name + ".smesh");
+	std::vector<MeshLib::Node> attr_pnts;
+	FileIO::TetGenInterface tgi;
+	bool result (tgi.writeTetGenSmesh(output_name, *mesh, attr_pnts));
+	ASSERT_TRUE(result);
+
+	GeoLib::GEOObjects geo_objects;
+	result = tgi.readTetGenGeometry(output_name, geo_objects);
+	ASSERT_TRUE(result);
+	std::string const ref_name(BaseLib::BuildInfo::data_path + "/FileIO/AmmerSubsurfaceCoarse.smesh");
+	result = tgi.readTetGenGeometry(ref_name, geo_objects);
```

Comment:
I wonder why `readTetGenGeometry` is called here a second time. The first call is in line 60. Can you comment this?